### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by natalia-se**
Added API to create, get and delete comments to the article.

### CODE REVIEW

File: middleware/auth.js [hunk 1]
```diff
@@ -10,4 +10,12 @@ const authorizeUser = (req, res, next) => {
...
 }
 
-module.exports = { authorizeUser }
+const requireLogin = (req, res, next) => {
+  if (req.user) {
+    next()
+  } else {
+    res.status(401).json("Login required!")
+  }
+}
```
Comment (imakero) [inline middleware/auth.js:19]: Magnificent!

Comment (imakero) [inline middleware/auth.js:19]: Magnificent!

File: routers/articles.js [hunk 2]
```diff
@@ -162,12 +164,86 @@ router.get("/:slug", async (req, res) => {
...
+// Get comments
+router.get("/:slug/comments", async (req, res) => {
+  const { slug } = req.params
+  const comments = await Comment.find({ slug })
+
+  if (!comments) {
+    return res.status(404).json({ success: false, msg: "No comments found." })
+  }
+
+  res.json({ comments })
```
Comment (imakero) [inline routers/articles.js:188]: Change `_id` to `id` before sending response!

Comment (imakero) [inline routers/articles.js:188]: Change `_id` to `id` before sending response?

Comment (natalia-se) [reply routers/articles.js:188]: Fixed

**Pull request merged**